### PR TITLE
chore: rename `hilla-file-router` to `file-router`

### LINKED ISSUE #2260: File router TS package is in a directory whose name does not follow existing pattern.
`hilla-file-router` should be renamed to `file-router`.

### PR DESCRIPTION
**Pull request by cromoteca**
Fixes #2260

### CODE REVIEW

File: packages/java/endpoint/src/main/java/com/vaadin/hilla/route/records/ClientViewConfig.java [hunk 1]
```diff
@@ -13,9 +13,6 @@
 /**
  * Implementation of TypeScript's Hilla ConfigView. Represents a view
  * configuration from Hilla file-system-routing module.
- *
- * @see <a href=
- *      "https://github.com/vaadin/hilla/tree/main/packages/ts/hilla-file-router/src/utils.ts#L3">ConfigView</a>
```
Comment (cromoteca) [inline packages/java/endpoint/src/main/java/com/vaadin/hilla/route/records/ClientViewConfig.java:18]: Link dead before PR

File: packages/java/endpoint/src/main/java/com/vaadin/hilla/route/records/ClientViewMenuConfig.java [hunk 2]
```diff
@@ -6,7 +6,6 @@
  * Represents a view's menu configuration
  * from Hilla file-system-routing.
  * Used for configuring MainLayout navigation items
- * @see <a href="https://github.com/vaadin/hilla/tree/main/packages/ts/hilla-file-router/src/utils.ts#L38">ConfigView menu</a>
```
Comment (cromoteca) [inline packages/java/endpoint/src/main/java/com/vaadin/hilla/route/records/ClientViewMenuConfig.java:9]: Link dead before PR

**Review by taefi — APPROVED** [review]
LGTM. Having uniform module naming that follows the same pattern (not having `hilla-` in their directory name) is important. The counter argument of having same module name as the directory name can also be valid, but having uniformity is more important IMO.

**Review by taefi — APPROVED** [review]
LGTM

**Pull request merged**